### PR TITLE
use the auto-expand depth option to expand up to 2 levels

### PR DESCRIPTION
**Pull request by nightpool**
depends on https://github.com/AprilSylph/JSONdrop/pull/2

### CODE REVIEW

**Review by AprilSylph — CHANGES_REQUESTED** [review]
i feel like the autoOpenDepth should be a theme option rather than a fixed value, dunno if there'd be any point going up to 3 but id definitely want 0 and 1 to be options without going into the theme code

**Comment by nightpool** [comment]
+1

File: Theme.html [hunk 1]
```diff
@@ -5,6 +5,7 @@
 		<link rel="stylesheet" href="https://static.tumblr.com/2wliono/oG8ps6bg4/jsondrop.css">
 		<link rel="stylesheet" href="https://static.tumblr.com/2wliono/PXMps6cfk/colour.css">
 		<script src="https://static.tumblr.com/zctqgvy/TaXps6sqb/jsondrop.js"></script>
+		<meta name="text:Auto-Expand Depth" content="2"/>
```
Comment (AprilSylph) [inline Theme.html]: i was thinking of a `select:` type option for foolproofing, but this does give more freedom...

Comment (nightpool) [reply Theme.html]: if you want foolproofing don't use a json theme

Comment (AprilSylph) [reply Theme.html]: fair

File: Theme.html [hunk 2]
```diff
@@ -4,7 +4,8 @@
 		<title>Theme JSON</title>
 		<link rel="stylesheet" href="https://static.tumblr.com/2wliono/oG8ps6bg4/jsondrop.css">
 		<link rel="stylesheet" href="https://static.tumblr.com/2wliono/PXMps6cfk/colour.css">
-		<script src="https://static.tumblr.com/2wliono/jGHps6w0v/jsondrop.js"></script>
+		<script src="https://static.tumblr.com/2wliono/I75ps707w/jsondrop.js"></script>
+		<meta name="text:Auto-Expand Depth" content="2"/>
```
Comment (AprilSylph) [inline Theme.html]: can't get this work, looks like tumblr theme variable names cant have hyphens :(
`"text:AutoExpand Depth"` works fine though

File: Theme.html [hunk 3]
```diff
@@ -884,7 +885,11 @@
 	</head>
 	<body>
 		<script>
-			document.body.innerHTML = `<div>${translate("", data)}</div>`;
+			document.body.innerHTML = `<div>${
+			    render(data, {
+					autoOpenDepth: {text:AutoExpand Depth},
```
Comment (AprilSylph) [inline Theme.html:891]: this whole line shouldn't be rendered if the option is left blank, i.e. should be wrapped in `{block:IfAutoExpandDepth}` `{/block:IfAutoExpandDepth}`

bad inputs breaking things is fine but blank inputs breaking things makes me sad

**Review by AprilSylph — CHANGES_REQUESTED** [review]

**Comment by nightpool** [comment]
the default is 2 though? so if you leave it blank shouldn't it return 2?

**Comment by AprilSylph** [comment]
nope, leaving a theme option blank makes it render blank. the default is just the default input, which the user can change

**Review by AprilSylph — APPROVED** [review]

**Pull request merged**